src/components/LocationPermissionModal/index.tsx

```diff
@@ -19,12 +22,43 @@ function LocationPermissionModal({startPermissionFlow, resetPermissionFlow, onDe
 
     const isWeb = getPlatform() === CONST.PLATFORM.WEB;
 
+    const checkPermission = useCallback(() => {
+        getLocationPermission().then((status) => {
+            if (status !== RESULTS.GRANTED && status !== RESULTS.LIMITED) {
+                return;
+            }
+            onGrant();
+        });
+    }, [onGrant]);
+
+    const debouncedCheckPermission = useMemo(() => lodashDebounce(checkPermission, CONST.TIMING.USE_DEBOUNCED_STATE_DELAY), [checkPermission]);
+
+    useEffect(() => {
+        if (!showModal) {
+            return;
+        }
+
+        const unsubscriber = Visibility.onVisibilityChange(() => {
+            debouncedCheckPermission();
+        });
+
+        const intervalId = setInterval(() => {
+            debouncedCheckPermission();
+        }, CONST.TIMING.LOCATION_UPDATE_INTERVAL);
```

Comment:
So we're polling instead of triggering the check on re-focus?

Reply:
@luacmartins we are combining polling, focus, and also re-checking the location permission when clicking `Continue` button. Polling also useful in case user not directly get back to the expensify app. So these combination make the detection more reliable

In chrome, onVisibilityChanged doesn't trigger after clicking the chrome's permission prompt.